### PR TITLE
Clueboard refresh

### PR DESCRIPTION
**Pull request by skullydazed**
## Description
Update Clueboard to follow modern practices.

Special thanks to @noroadsleft for doing the initial cleanup work.

## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [x] Keyboard (addition or update)
- [x] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* 

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/clueboard/66/rev2/config.h [hunk 1]
```diff
@@ -1,7 +1,4 @@
-#ifndef CLUEBOARD_66_REV2_CONFIG_H
-#define CLUEBOARD_66_REV2_CONFIG_H
-
-#include "config_common.h"
+#pragma once
```
Comment (noroadsleft) [inline keyboards/clueboard/66/rev2/config.h]: Line 38 of this file needs to be deleted.

File: keyboards/clueboard/17/17.h [hunk 2]
```diff
@@ -19,18 +19,21 @@
...
-    { k30, k31, k32, k33, }, \
-    { k40, KC_NO, k42, KC_NO } \
+    { k00, k01,   k02, k03   }, \
+    { k10, k11,   k12, k13   }, \
+    { k20, k21,   k22, KC_NO }, \
+    { k30, k31,   k32, k33   }, \
+    { k40, KC_NO, k42, KC_NO }  \
 }
 
+#define LAYOUT LAYOUT_numpad_5x4
```
Comment (drashna) [inline keyboards/clueboard/17/17.h]: /cries

File: keyboards/clueboard/2x1800/rules.mk [hunk 3]
```diff
@@ -1,64 +1,21 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
 MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
 UNICODE_ENABLE = no         # Unicode
-RGBLIGHT_ENABLE = no        # RGB on port C6
-AUDIO_ENABLE = no           # Audio output on port C4 and B7
-FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+RGBLIGHT_ENABLE = yes       # RGB on port C6
+AUDIO_ENABLE = yes          # Audio output on port C4 and B7
+FAUXCLICKY_ENABLE = yes     # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/clueboard/2x1800/rules.mk]: This needs to be turned off.
```suggestion
FAUXCLICKY_ENABLE = no   # Use buzzer to emulate clicky switches
```

You'd want to add `#define AUDIO_CLICKY` in the config.h to enable the AUDIO version.

File: keyboards/clueboard/66/config.h [hunk 4]
```diff
@@ -15,10 +15,7 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#ifndef CLUEBOARD_66_CONFIG_H
-#define CLUEBOARD_66_CONFIG_H
-
-#include "config_common.h"
```
Comment (drashna) [inline keyboards/clueboard/66/config.h]: I think you still need this, actually.

```suggestion
#include "config_common.h"	
```

File: keyboards/clueboard/66_hotswap/gen1/config.h [hunk 5]
```diff
@@ -1,5 +1,4 @@
 #pragma once
-#include "config_common.h"
```
Comment (drashna) [inline keyboards/clueboard/66_hotswap/gen1/config.h]: This is still needed. 

```suggestion
#include "config_common.h"
```

File: keyboards/clueboard/66_hotswap/config.h [hunk 6]
```diff
@@ -16,7 +16,6 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #pragma once
-#include "config_common.h"
```
Comment (drashna) [inline keyboards/clueboard/66_hotswap/config.h]: This is still needed. 

```suggestion
#include "config_common.h"
```

File: keyboards/clueboard/60/config.h [hunk 7]
```diff
@@ -26,29 +26,36 @@
 #define PRODUCT         Clueboard 60%
```
Comment (drashna) [inline keyboards/clueboard/60/config.h]: This file needs the following somewhere: 

```c
#include "config_common.h"
```

File: keyboards/clueboard/66/rules.mk [hunk 8]
```diff
@@ -1,4 +1,3 @@
-DEFAULT_FOLDER = clueboard/66/rev2
```
[the remark this answers is not in the document]
Comment (skullydazed) [reply keyboards/clueboard/66/rules.mk]: Yeah, I would have already pushed this change but something is broken and I can't figure out what yet. :/

File: keyboards/clueboard/card/config.h [hunk 9]
```diff
@@ -15,10 +15,7 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#ifndef CONFIG_H
-#define CONFIG_H
-
-#include "config_common.h"
```
Comment (drashna) [inline keyboards/clueboard/card/config.h]: This is still needed. 

```suggestion
#include "config_common.h"
```

File: keyboards/clueboard/66/rules.mk (hunk 8 above)
[the remark this answers is not in the document]
Comment (skullydazed) [reply keyboards/clueboard/66/rules.mk]: 1dd51c55f fixes this.

File: keyboards/clueboard/66/rev2/config.h (hunk 1 above)
Comment (skullydazed) [reply keyboards/clueboard/66/rev2/config.h]: 1dd51c55f fixes this

File: keyboards/clueboard/2x1800/rules.mk (hunk 3 above)
Comment (drashna) [reply keyboards/clueboard/2x1800/rules.mk]: @noroadsleft Yeah, Fauxclicky the older method that was hard coded to pin C6.  It also had some other issues with using it to.

I believe it was mindsound that wrote the newer code, as part of a keymap, and I ported that to the core functionality.  

Fauxclicky is smaller, (even with music mode turned off).

**Comment by skullydazed** [comment]
The only remaining error is a firmware too large, and that's in a user's keymap so I'm not going to touch it for now.

**Pull request merged**